src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Testing/RunTestsHandler.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Composition;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.Testing;
+using Microsoft.TestPlatform.VsTestConsole.TranslationLayer;
+using Roslyn.Utilities;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Testing;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(RunTestsHandler)), Shared]
+[Method(RunTestsMethodName)]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal class RunTestsHandler(DotnetCliHelper dotnetCliHelper, TestDiscoverer testDiscoverer, TestRunner testRunner, ServerConfiguration serverConfiguration) : ILspServiceDocumentRequestHandler<RunTestsParams, RunTestsPartialResult[]>
+{
+    private const string RunTestsMethodName = "textDocument/runTests";
+
+    public bool MutatesSolutionState => false;
+
+    public bool RequiresLSPSolution => true;
+
+    public LSP.TextDocumentIdentifier GetTextDocumentIdentifier(RunTestsParams request)
+    {
+        return request.TextDocument;
+    }
+
+    public async Task<RunTestsPartialResult[]> HandleRequestAsync(RunTestsParams request, RequestContext context, CancellationToken cancellationToken)
+    {
+        Contract.ThrowIfNull(context.Document);
+        using var progress = BufferedProgress.Create(request.PartialResultToken);
+
+        // First, build to make sure we have a relatively up to date project.
+        await BuildAsync(context.Document, progress, cancellationToken);
```

Comment:
ConfigureAwait?

concerned that this didn't error in the analyzer.

Reply:
applies to all awaits.

Reply:
the server executable project doesn't require it (since it doesn't make a difference there).

Reply:
ok.  i can sorta live with it.  but i think i've seen some CA's in the code.  so the inconsistency bums me out :)  consider making it a banned api in this project :)